data/src/test/java/org/apache/iceberg/data/DataTestHelpers.java

```diff
@@ -36,7 +36,10 @@ public static void assertEquals(Types.StructType struct, Record expected, Record
         assertEquals(
             field.type(), expected.getField(expectedField.name()), actual.getField(field.name()));
       } else {
-        assertThat(actual.getField(field.name())).isEqualTo(field.initialDefault());
+        assertEquals(
+            field.type(),
+            GenericDataUtil.internalToGeneric(field.type(), field.initialDefault()),
+            actual.getField(field.name()));
```

Comment:
This was the test bug. Rather than asserting that the value was passed back unchanged, it should have delegated to `assertEquals` that checks the representation of the value.

The expected value (default) also needs to be converted to generic because the generic data model tests use the generic data model as the source representation.